glusterd2/transaction/transaction.go

```diff
@@ -31,14 +32,9 @@ type Txn struct {
 }
 
 // NewTxn returns an initialized Txn without any steps
-func NewTxn(id string) *Txn {
+func NewTxn(ctx context.Context) *Txn {
 	t := new(Txn)
-
-	if t.ID = uuid.Parse(id); t.ID == nil {
-		t.ID = uuid.NewRandom()
-		log.WithField("reqid", t.ID.String()).Warn("Invalid UUID set as request ID. Generated new request ID")
-	}
-
+	t.ID = uuid.Parse(ctx.Value(gdctx.ReqIDKey).(string))
```

Comment:
With middlewares like the disk management plugin, we could have multiple transactions launched for a single request. Using req-id as transaction id doesn't make good sense in such a case. We should start using unique ids per transaction.

Reply:
I've had the same thought too. I'll do that separately and also have `txnid` derived from `reqid` or come up with some other mechanism to co-relate the two.